src/core/entities/crux-user.ts

```diff
@@ -0,0 +1,70 @@
+import { CruxId } from "src/packages/identity-utils";
+import { BaseError } from "../../packages/error";
+import { getLogger } from "../../packages/logger";
+import { CruxSpec } from "./crux-spec";
+
+const log = getLogger(__filename);
+
+export interface IAddress {
+    addressHash: string;
+    secIdentifier?: string;
+}
+
+export interface IAddressMapping {
+    [currency: string]: IAddress;
+}
+
+export interface ICruxUserRegistrationStatus {
+    status: SubdomainRegistrationStatus;
+    statusDetail: SubdomainRegistrationStatusDetail;
+}
+
+export enum SubdomainRegistrationStatus {
+    NONE = "NONE",
+    PENDING = "PENDING",
+    DONE = "DONE",
+    REJECT = "REJECT",
+}
+
+export enum SubdomainRegistrationStatusDetail {
+    NONE = "Subdomain not registered with this registrar.",
+    PENDING_REGISTRAR = "Subdomain registration pending on registrar.",
+    PENDING_BLOCKCHAIN = "Subdomain registration pending on blockchain.",
+    DONE = "Subdomain propagated.",
+}
+
+export class CruxUser {
+    public cruxID: CruxId;
```

Comment:
even cruxID should be behind a getter as setting a cruxID needs protection.

Reply:
same applies for the cruxDomainId in CruxDomain.

Reply:
Made the variable and behind getter as suggested